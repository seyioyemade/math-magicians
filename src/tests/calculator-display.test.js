import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CalculatorDisplay from '../components/calculator-display';

describe('Calculator Display', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BrowserRouter><CalculatorDisplay /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
