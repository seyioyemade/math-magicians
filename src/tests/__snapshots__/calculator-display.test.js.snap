// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Display renders correctly 1`] = `
<section>
  <table>
    <tr>
      <th
        colSpan="4"
      >
        0
      </th>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td
        colSpan="2"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </td>
      <td>
        <button
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </td>
    </tr>
  </table>
</section>
`;
